components/src/directives/focus-first-element/index.ts

```diff
@@ -9,28 +9,43 @@ const excludeElements =
   'button:not(.oxd-dialog-close-button,.modal-reset-button)';
 let firstFocusableElement: Element;
 let updatedFocusFirstElement: Element;
+const firstFocusedElementsOnMounted = new Map();
+const firstFocusedElementsOnUpdated = new Map();
 
-const focusOnFirstElement = (element: Element, matchingString: string) => {
+const focusOnFirstElement = (
+  element: Element,
+  matchingString: string,
+  vnode: VNode,
+) => {
   firstFocusableElement = element.querySelectorAll(matchingString)[0];
+  firstFocusedElementsOnMounted.set(vnode.scopeId, firstFocusableElement);
   if (firstFocusableElement) {
     (firstFocusableElement as HTMLElement).focus();
   }
 };
 
 const focusonFirstElementDirective: Directive = {
-  updated(el: FocusFirstHTMLElement) {
+  updated(el: FocusFirstHTMLElement, binding, vnode: VNode) {
     updatedFocusFirstElement = el.querySelectorAll(
       focusableElements + ', ' + excludeElements,
     )[0];
-    if (firstFocusableElement !== updatedFocusFirstElement) {
-      focusOnFirstElement(el, focusableElements + ', ' + excludeElements);
+    firstFocusedElementsOnUpdated.set(vnode.scopeId, updatedFocusFirstElement);
```

Comment:
```
 updated(el: FocusFirstHTMLElement, binding, vnode: VNode) {
  const elm = el.querySelectorAll(focusableElements + ', ' + excludeElements)[0];
  firstFocusedElementsOnUpdated.set(vnode.scopeId, elm);
  if (
    firstFocusedElementsOnMounted.has(vnode.scopeId) &&
    elm !== firstFocusedElementsOnUpdated.get(vnode.scopeId)
  ) {
    focusOnFirstElement(
      el,
      focusableElements + ', ' + excludeElements,
      vnode,
    );
  }
};
```

Reply:
guess here the suggested comparison is invalid as need to compare between mounted and updated element of same scopeId.